GETTING_STARTED.md

```diff
@@ -0,0 +1,192 @@
+need to add some basic type to Silver, or you need to add some kind of
+function on basic types, it goes in here.
+
+
+
+### Silver's Definition
+
+The host language of Silver is in
+`grammars/silver/compiler/definition`.
+
+
+The `core` directory in this directory is the home of the definitions
+of most of the constructs in Silver.  For example, most Silver
+host-language expressions are defined in `Expr.sv` here.
+
+
+Types are defined in the `type` directory under `definition`.  Types
+are assigned to expressions in `definition/core`, but type checking to
+generate error messages is done in
+`grammars/silver/compiler/analysis/typechecking/core`.  This
+separation is probably a bad choice, but what is done is done.
+
+
+
+### Extensions
+
+Extensions to Silver go in `grammars/silver/compiler/extension`.  Each
+extension has its own directory here which contains the code for the
+extension.
+
+
+The extension must also be added to the list of extensions in
+`grammars/silver/compiler/host/Project.sv`.  This means it will be
+available as part of Silver; without this, it won't be included in
+Silver.
+
+
+
+### Modifications
+
+Modifications are things which aren't conceptually part of the Silver
+host language, but which can't be pure extensions.  For Silver, these
+include lambda functions and collection attributes.  They can be found
+in `grammars/silver/compiler/modification`.  Rarely should a construct
+be added as a modification.
+
+
+
+
+## Making Changes
+
+You should always make a new branch when making changes to Silver.
+**Never** make changes directly on the `develop` branch.  Your
+branch's name should take one of two forms, depending on what kind of
+changes you are making:
+* If you are fixing something so it works as intended, your branch's
+  name should start with `bugfix/` followed by a descriptive name.
+  For example, `bugfix/list-append-crash` would be a branch fixing the
+  behavior of appending lists if it sometimes caused Silver to crash.
+* If you are adding a new feature to Silver, your branch's name should
+  start with `feature/` followed by a descriptive name.  For example,
+  `feature/list-subtraction` would be a branch adding the ability to
+  write code like `[1, 2] - [1]`.
+
+You can add your branch to the remote Silver repository by
+```
+git push -u origin <branch>
+```
+
+
+When you are making changes, you should add test cases for them.
+These go in the `test` directory, which has more directories in it for
+organizing test cases.  Unless there is already another directory for
+testing related to what you are working on (e.g. pattern matching
+tests go in `test/patt`), it should go in `test/silver_features`.  The
+syntax for writing tests is best seen by looking at other test files.
+
+
+
+
+## Helpful Hints
+
+A good way to see where something (e.g. a function) is used in the
+Silver code base is to use the following command:
+```
+grep -r --include="*.sv" "function name"
```

Comment:
I typically just use `git grep functionName` as that is quicker to type (and also shows up references in the java runtime.)  

Also there are some similar tools under `support/bin/scripts`.